File: Java/lab22_Template_Method/task_2_1 – solution/src/VegetarianPizza.java
public class VegetarianPizza extends Pizza {

  @Override
  public void cook() {
    System.out.println("Cook in the oven for 15 minutes");
  }

  public void addToppings() {
    super.addToppings();
    System.out.println("Add peppers");
    System.out.println("Add olives");
  }






}

File: Cpp/lab22_Template_Method/task_2_1 - solution/src/vegetarian_pizza.cpp
#include <iostream>

#include "vegetarian_pizza.hpp"


void VegetarianPizza::addToppings()
{
  this->Pizza::addToppings();
  std::cout << "Add peppers" << std::endl
            << "Add olives" << std::endl;
}

void VegetarianPizza::cook()
{
    std::cout << "Cook in the oven for 15 minutes" << std::endl;
}
